lib/membrane_file/common_file.ex

```diff
@@ -1,26 +1,47 @@
 defmodule Membrane.File.CommonFile do
   @moduledoc false
 
-  def open(location \\ nil, mode, state) do
-    location = location || state.location
-    path = Path.expand(location)
+  alias Membrane.{Buffer, Payload}
+  alias Membrane.File.{Error, SeekEvent}
 
-    with {:ok, fd} <- File.open(path, [:binary, mode]) do
-      {:ok, %{state | fd: fd}}
+  @spec open(String.t(), [File.mode() | :ram]) :: {:ok, File.io_device()} | Error.posix_error_t()
+  def open(location, modes), do: File.open(Path.expand(location), [:binary | List.wrap(modes)])
+
+  @spec write(File.io_device(), Buffer.t()) :: :ok | Error.posix_error_t()
+  def write(fd, %Buffer{payload: payload}), do: IO.binwrite(fd, Payload.to_binary(payload))
+
+  @spec seek(File.io_device(), SeekEvent.position_t()) :: :ok | Error.generic_error_t()
+  def seek(fd, position), do: :file.position(fd, position)
+
+  @spec copy(File.io_device(), File.io_device()) :: :ok | Error.generic_error_t()
```

Comment:
Invalid spec,  You are returning a tuple of `{:ok, bytes_copied}` instead of `:ok` atom.